### ISSUE TITLE
Format: Add support for CSV files

### ISSUE THREAD

**Issue by pedro-mendonca** [GlotPress/GlotPress#820]
Hi,

I would like to request the addition of CSV file format to the export options.
Is it possible?

Thanks

**Comment by ocean90** [comment]
Sounds interesting but I don't think that CSV is a good format for translations if you think of all the singular/plural handling. Do you have a specific use case in mind?

**Comment by pedro-mendonca** [comment]
Yes, I use it myself for later import to TablePress for a translation search tool.
The plural is handled currently as a two line value, works fine, but I have to do it through some manual steps.
Example: wp-portugal.com/traducao/tabela/
The plural handling might be an issue for a bundled feature, maybe it can be done through an independent plugin.

**Comment by toolstack** [comment]
Is there a standard that TablePress uses for the cvs files published somewhere?

**Comment by pedro-mendonca** [comment]
It tries to guess the imported file format:

> **CSV (Character-separated Values)**
>     Every row is in a new line, every column is separated by a character (like “;” (semicolon), “,” (comma), or “\tab” (tabulator)). The plugin will try to determine the used separation character automatically. See the Wikipedia article for more information on the CSV format.

https://tablepress.org/faq/documentation-import-export-formats/

**Comment by ePascalC** [comment]
I love TablePress, really! The way of how you can show tables with all sorting, filtring, headers, etc is just great. But I do not see it as a use case for exporting csv from GlotPress.

That said, I FULLY support the request for a CSV export. As plugin author I would just love to get a CSV file, import it into Excel or other and see where I'm missing context or comments.
I also saw over the years requests like 'how many words are there in this plugin', so exporting to CSV and then use calculations in any spreadsheet might be more accurate/easier then use tools to calculate from the PO.

As you see I only talk about 'export' because I think import should be done in known formats for strings/translations. But if you think you can also import, then why not!

![glotpresscsv](https://user-images.githubusercontent.com/18325208/30363108-43c63080-985f-11e7-819d-0669cfc141d6.PNG)

**Comment by ePascalC** [comment]
CSV standard: https://tools.ietf.org/html/rfc4180

**Comment by pedro-mendonca** [comment]
@ePascalC 

> But I do not see it as a use case for exporting csv from GlotPress.

I agree, I'll probably go for an independent solution for my specific case, still, CSV would be a great adition as a generic export.

**Comment by ePascalC** [comment]
So I would say 'add it' !

**Comment by pedro-mendonca** [comment]
After a few years, I'm picking this old issue to take a decision.

I'm using a solution that deals with the hard singular/plurals as described below.
The default CSV columns are:
- Original Singular
- Original Plural
- Context
- Comments (single string with all the comments as extracted from the Translation entry, separated by `\n`)
- For `GP_Locale->nplurals` = 2:
  - Translation Singular
  - Translation Plural
- For other `GP_Locale->nplurals`:
  - Translation for X
  - Translation for Y
  - (etc)

As the translations are the last columns of the CSV, their number may vary without breaking or needing any custom mapping.

As an extra, this column list could be filterable, as well as the line content, allowing any export customizations.

I want the team's feedback about if it's a good thing to include in the core, or I'll just close this issue and keep it as a plugin.

**Comment by pedro-mendonca** [comment]
Hi again, I've made a [plugin](https://github.com/pedro-mendonca/GP-Format-CSV) for this, @ePascalC you can test on your workflow and let me know if it works for you. It exports and imports all the necessary fields both for originals and translations.

**Comment by psmits1567** [comment]
If there is a solution can we close this issue??

**Comment by pedro-mendonca** [comment]
I'm closing this as I was the author of the request and I already have a plugin solution.
